PostHog/Autocapture/SwiftUI/View+PostHogLabel.swift

```diff
@@ -0,0 +1,141 @@
+//
+//  View+PostHogLabel.swift
+//  PostHog
+//
+//  Created by Yiannis Josephides on 04/12/2024.
+//
+
+#if os(iOS) || targetEnvironment(macCatalyst)
+    import SwiftUI
+
+    public extension View {
+        /**
+         Adds a custom label to this view for use with PostHog's auto-capture functionality.
+
+         By setting a custom label, you can easily identify and filter interactions with this specific element in your analytics data.
+
+         ### Usage
+         ```swift
+         struct ContentView: View {
+            var body: some View {
+                Button("Login") {
+                    ...
+                }
+                .postHogLabel("loginButton")
+              }
+         }
+         ```
+
+         - Parameter label: A custom label that uniquely identifies the element for analytics purposes.
+         */
+        func postHogLabel(_ label: String?) -> some View {
+            modifier(PostHogLabelTaggerViewModifier(label: label))
+        }
+    }
+
+    private struct PostHogLabelTaggerViewModifier: ViewModifier {
+        let label: String?
+
+        func body(content: Content) -> some View {
+            content
+                .background(PostHogLabelViewTagger(label: label))
+        }
+    }
+
+    private struct PostHogLabelViewTagger: UIViewRepresentable {
+        let label: String?
+
+        func makeUIView(context _: Context) -> PostHogLabelTaggerView {
+            PostHogLabelTaggerView(label: label)
+        }
+
+        func updateUIView(_: PostHogLabelTaggerView, context _: Context) {
+            // nothing
+        }
+    }
+
+    private class PostHogLabelTaggerView: UIView {
+        private let label: String?
+
+        init(label: String?) {
+            self.label = label
+            super.init(frame: .zero)
+        }
+
+        @available(*, unavailable)
+        required init?(coder _: NSCoder) {
+            fatalError("init(coder:) has not been implemented")
+        }
+
+        override func layoutSubviews() {
+            super.didMoveToWindow()
+
+            superview?.postHogNoCapture = true
+            postHogNoCapture = true
```

Comment:
I am a bit confused, why are we doing this here if this modifier is about the view's id?